clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1085,6 +1085,64 @@ def CmpOp : CIR_Op<"cmp", [Pure, SameTypeOperands]> {
   let hasVerifier = 0;
 }
 
+//===----------------------------------------------------------------------===//
+// CheckedBinOp
+//===----------------------------------------------------------------------===//
+
+def CheckedBinOpKind : I32EnumAttr<
+    "CheckedBinOpKind",
+    "checked binary arithmetic operation kind",
+    [BinOpKind_Add, BinOpKind_Sub, BinOpKind_Mul]> {
+  let cppNamespace = "::mlir::cir";
+}
+
+def CheckedBinOp : CIR_Op<"binop.overflow", [Pure, SameTypeOperands]> {
```

Comment:
nit: the actual MLIR names should match the asm one:
`CheckedBinOp` -> `BinOpOverflow`
`CheckedBinOpKind` -> `BinOpOverflowKind`

Reply:
Updated.